### PR TITLE
upgrade: ⬆️ PROS 4.1.0

### PR DESCRIPTION
**Pull request by jmakif**
### Summary
 This pull request updates the project to use PROS 4.1, making any necessary code fixes (with no changes in functionality), and updates the tutorials accordingly.

## Download the template for this pull request: 

> [!NOTE]  
> This is auto generated from [`Add Template to Pull Request`](https://github.com/LemLib/LemLib/actions/runs/9148923908)
- via manual download: [LemLib@0.5.0+47a320.zip](https://nightly.link/LemLib/LemLib/actions/artifacts/1517243885.zip)
- via PROS Integrated Terminal: 
 ```
curl -o LemLib@0.5.0+47a320.zip https://nightly.link/LemLib/LemLib/actions/artifacts/1517243885.zip;
pros c fetch LemLib@0.5.0+47a320.zip;
pros c apply LemLib@0.5.0+47a320;
rm LemLib@0.5.0+47a320.zip;
```

### CODE REVIEW

**Comment by ABUCKY0** [comment]
Did you update the documentation for any of the PROS 4 changes?

File: .gitignore
Comment (liam-teale) [inline .gitignore:1]: the gitignore file got nuked by the pros cli

File: docs/tutorials/2_configuration.md [hunk 1]
```diff
@@ -45,47 +41,23 @@ This needs to be done for all motors on the drivetrain
...
 Now, we need to specify what cartridge is used by every motor. The cartridge can be checked by looking at the area below the shaft of the motor. A motor can have one of three cartridges:
 
 |          **Color**         |         **RPM**         | **Code Representation**    |
 | -------------------------- | ----------------------- | -------------------------- | 
-| **Red**                    |           100           | `pros::E_MOTOR_GEAR_RED` |
-| **Green**                  |           200           | `pros::E_MOTOR_GEAR_GREEN` |
-| **Blue**                   |           600           | `pros::E_MOTOR_GEAR_BLUE` |
+| **Red**                    |           100           | `pros::MotorGearset::red` |
+| **Green**                  |           200           | `pros::MotorGearset::green` |
+| **Blue**                   |           600           | `pros::MotorGearset::blue` |
```
Comment (liam-teale) [inline docs/tutorials/2_configuration.md]: `|` are not aligned

File: docs/tutorials/2_configuration.md [hunk 2]
```diff
@@ -45,47 +41,23 @@ This needs to be done for all motors on the drivetrain
...
-pros::Motor front_right_motor(4, pros::E_MOTOR_GEAR_GREEN); // right_motor_group
-pros::Motor middle_right_motor(5, pros::E_MOTOR_GEAR_BLUE); // right_motor_group
-pros::motor back_right_motor(6, pros::E_MOTOR_GEAR_RED); // right_motor_group
-
-// left motor group
-pros::MotorGroup left_motor_group({ front_left_motor, middle_left_motor, back_left_motor });
-// right motor group
-pros::MotorGroup right_motor_group({ front_right_motor, middle_right_motor, back_right_motor });
+pros::MotorGroup left_motors({-1, 2, -3}, pros::MotorGearset::blue); // left motors use 600 RPM cartrifges
+pros::MotorGroup right_motors({4, -5, 6}, pros::MotorGearset::green); // right motors use 200 RPM cartridges
```
Comment (liam-teale) [inline docs/tutorials/2_configuration.md:60]: add an admonition stating what needs to be done if motors on one side of the drive have different motor cartridges

**Comment by BattleCh1cken** [comment]
> Did you update the documentation for any of the PROS 4 changes?

Yes, everything looks good on my end.

File: src/lemlib/chassis/chassis.cpp [hunk 3]
```diff
@@ -55,7 +54,7 @@ void calibrateIMU(lemlib::OdomSensors& sensors) {
         sensors.imu->reset();
         // wait until IMU is calibrated
         do pros::delay(10);
-        while (sensors.imu->get_status() != 0xFF && sensors.imu->is_calibrating());
+        while (sensors.imu->get_status() != pros::ImuStatus::error && sensors.imu->is_calibrating());
```
Comment (liam-teale) [inline src/lemlib/chassis/chassis.cpp:57]: has this been tested? I don't trust it after PROS 3.8.2 and 3.8.3

**Comment by liam-teale** [comment]
Example code and tutorials need to be verified, as well as odom

File: src/lemlib/chassis/chassis.cpp (hunk 3 above)
Comment (liam-teale) [reply src/lemlib/chassis/chassis.cpp:57]: Turns out `pros::ImuStatus::error` is equal to `0xFF`, marking as resolved

File: docs/tutorials/2_configuration.md (hunk 2 above)
Comment (jmakif) [reply docs/tutorials/2_configuration.md:60]: PROS 4.1 still doesn't have the ability to put individual motors in motorgroups, so I'm gonna resolve this

**Pull request merged**